### DISCUSSION TITLE
Recommended pattern for integrated Adiak+Caliper builds

### DISCUSSION THREAD

**Discussion by DavidPoliakoff** [LLNL/Caliper#338] [Q&A]
I figure people have to do this, wanted to get a best practice.

I want to build Adiak-enabled Caliper as a subcomponent of a tool. Right now Caliper finds Adiak (find_package). Do you have a pattern to have a structure like

```
project_root
  tpl
    caliper
    adiak
```

And get a Caliper-enabled Adiak I can link? Right now it looks like static-build Adiak with a custom command in CMake, then add Caliper pointing at the now-built Adiak. Or are there plans to do with Adiak what was done with Gotcha? Or use submodules?

What's a best practice, here?

**Comment by daboehme** [comment]
I don't think we have a "standard procedure" or best practice for building Caliper and Adiak together, other than maybe using spack. Applications all do their own thing it seems.

We don't currently plan to package adiak with Caliper like we do with gotcha either: unlike gotcha the application itself also uses adiak, and if the application wants to use one version of adiak but Caliper ships another, you can run into weird issues with the compiler picking the "wrong" version depending on the order of include directories.
